src/unsync/rwlock.rs

```diff
@@ -0,0 +1,395 @@
+use core::cell::Cell;
+
+use lock_api::{
+    GuardSend, RawRwLock, RawRwLockDowngrade, RawRwLockRecursive, RawRwLockUpgrade,
+    RawRwLockUpgradeDowngrade,
+};
+
+/// A one-shot readers-writer lock that panics instead of (dead)locking on contention.
+///
+/// This lock allows no contention and panics on [`lock_shared`], [`lock_exclusive`], [`lock_upgradable`], and [`upgrade`] if it is already locked conflictingly.
+/// This is useful in situations where contention would be a bug,
+/// such as in single-threaded programs that would deadlock on contention.
+///
+/// This lock does not implement `Sync`, which permits a slightly more efficient implementation.
+/// For a variant that does implement `Sync`, see [`RawOneShotRwLock`](crate::RawOneShotRwLock).
+///
+/// [`lock_shared`]: RawOneShotRwLock::lock_shared
+/// [`lock_exclusive`]: RawOneShotRwLock::lock_exclusive
+/// [`lock_upgradable`]: RawOneShotRwLock::lock_upgradable
+/// [`upgrade`]: RawOneShotRwLock::upgrade
+///
+/// # Examples
+///
+/// ```
+/// use one_shot_mutex::OneShotRwLock;
```

Comment:
This should also use the unsync variant.

Reply:
Done.